Feature/professor page

euny0ung opened the pull request:
참여교수 페이지를 구현했습니다

![참여교수 완성](https://user-images.githubusercontent.com/56199780/168065094-34adb86a-8615-441b-ba3c-c9394912eb38.PNG)

euny0ung committed:
fix: Update Prof

로딩 오류를 수정하였습니다.
#20

baaaam771 committed:
fix : Add header,subheader, footer

component 폴더 안에 있는 헤더, 서브해더, 푸터 경로참조하여 불러옴

[#20] issue 20-4

euny0ung committed:
fix: Update Prof
이미지 경로 오류를 수정하였습니다.
- 기존 이미지 삭제 후 랜덤으로 이미지를 보여주는 사이트 사용
#20

euny0ung committed:
refactor: Update Prof
필요없는 코드를 삭제하고 정리하였습니다.
-import를 간단하게 변경
-Card 태그의 width="50%" 삭제
-marginLeft로 정렬 -> Grid로 변경
#20

euny0ung committed:
style: Update Prof

텍스트, 컴포넌트 위치와 크기 변경
- 가독성을 위해 이름과 정보 사이를 한칸 띄웠습니다
- 글자 크기를 크게 하였습니다
- 컴포넌트가 중앙에 위치하도록 변경하였습니다
- 창 크기에 따라 반응하도록 변경하였습니다

#20

euny0ung committed:
Merge branch 'develop' into feature/professor_page

LimEunSang commented on src/pages/member/Prof.js:
```diff
@@ -0,0 +1,170 @@
+                  alt="image"
+                />
+                {/* 텍스트 정보 */}
+                <Box
```
이 Box는 아무런 역할을 안 하고 있습니다. 지워도 결과는 똑같은데 Box tag를 쓴 이유가 따로 있나요??

euny0ung replied:
이전 코드와 비교해보니 Box를 Grid로 변경하는 과정에서 실수로 sx만 주석처리하고 삭제하지 않은 것 같습니다. 올리기 전에 꼼꼼하게 검토하겠습니다!

LimEunSang commented on src/pages/member/Prof.js:
```diff
@@ -0,0 +1,170 @@
+                      <strong>{element.name}</strong>
+                    </Typography>
+                    <br></br>
+                    <Typography
```
같은 속성을 갖는 Typography가 여러 번 등장하네요. 
유지보수를 용이하게 하고 코드의 길이를 줄이기 위해서 
이 속성을 갖는 Typography를 재정의 해서 쓰거나 또는 createTheme를 사용하면 더 좋을 거 같아요!

suinj8 replied:
저도 이방법이 좋아보입니다.

LimEunSang commented on src/pages/member/Prof.js:
```diff
@@ -0,0 +1,170 @@
+                    <Typography
+                      variant="body2"
+                      color="text.secondary"
+                      component="div"
```
component="div" 속성을 준 이유가 따로 있을까요??

suinj8 replied:
> component="div" 속성을 준 이유가 따로 있을까요??

div를 줘야 줄바꿈이 되는걸로 알고있습니다.
확실한 답변은 은영님이 해주시겠죠?

LimEunSang replied:
> div를 줘야 줄바꿈이 되는걸로 알고있습니다. 확실한 답변은 은영님이 해주시겠죠?

없어도 똑같더라구요

euny0ung replied:
일단 답변 늦어서 죄송합니다. 참고한 자료가 영어로 써있어서 읽는데 한참 걸렸습니다...
이 코드의 의도는 다음과 같습니다.

<div><Typography>{...}</Typography><div>
<div><Typography>{...}</Typography><div>
<div><Typography>{...}</Typography><div>

그런데 은상님이 말씀해주신대로 component="div"를 삭제해도 결과가 같더라구요.
mui 공식 문서에 검색을 해보니 Box의 경우, component="div"가 디폴트로 들어있다고 합니다.
( a <div> that by default can be changed with the component prop.)
공식 문서에 Typography 관련 내용은 나오지 않아 Box와 마찬가지로 div가 디폴트로 들어가있는게 아닐까 하고 넘어가려다가 찝찝해서 mui 이슈를 찾아보았습니다. 

[mui 이슈] https://github.com/mui/material-ui/issues/14721

제가 보기에는 은상님 질문과 상관이 없는 것 같은데 제 해석이 정확하지 않을 수 있으니 일단 링크를 올려둘게요. 해석 가능하신분이 읽고 저한테도 알려주시면 감사하겠습니다!
또, 공식은 아니지만 이런 글이 있긴합니다.
[스택오버플로우 Typography] https://stackoverflow.com/questions/59268156/when-should-typography-be-used-in-material-ui
결론은, Typography도 div 태그가 디폴트로 들어있는것 같다.. 입니다.

LimEunSang replied:
와... 완벽주의자다운 답변이네요. 확실히 이해했습니다..!

LimEunSang reviewed (commented):
11번째, 33번째 줄 article page 때문에 실행 오류가 발생합니다! 지워야 실행이 가능하고, 병합할 때 다시 추가해야 할 듯 합니다.

[창 크기를 줄였을 때]
![image](https://user-images.githubusercontent.com/86942472/168198252-df9a371b-c812-4137-b315-44aa1adf5b89.png)

창 크기 줄이면 사진이 과도하게 홀쭉해지는 문제가 있네요. 하지만 사진이 확정이 된 이후에 CardMedia 속성을 수정해야 해서 당장 큰 문제는 안 될 거 같아요. 해결하면 좋지만 여유가 안되면 이슈에 등록만 해놓고 머지를 우선 하죠!

창 크기에 따라 post 크기 조절하기 어려워하셨던 거 같은데 잘 해결하셨네요! 고생하셨어요!!

suinj8 commented on src/App.js:
```diff
@@ -24,6 +25,7 @@ const App = () => {
       <BrowserRouter>
         <Routes>
           <Route path="/" element={<Home />} />
+          <Route path="/professor" element={<Prof />} />
```
현재 교수페이지만 줄임말로 되어있는데 
다 쓰는게 통일성 있지 않을까요?

euny0ung replied:
수정하였습니다!

suinj8 reviewed (commented):
.

baaaam771 commented:
> 11번째, 33번째 줄 article page 때문에 실행 오류가 발생합니다! 지워야 실행이 가능하고, 병합할 때 다시 추가해야 할 듯 합니다.
> > [chore: Edit article path name](https://github.com/AI-Research-Center-Homepage/frontend/commit/0165d7824cdf2b5db49de43821b8ef3831f65d24) 방금 확인했습니다 dev 브랜치에 수정하여 게시했습니다!

baaaam771 commented on src/pages/member/Prof.js:
```diff
@@ -0,0 +1,170 @@
+                <CardMedia
+                  component="img"
+                  sx={{
+                    width: '20%',
```
축소할 때 사진 작아지는 것은
width : 150
정도로 고정하니 보기 좋아보이네요

baaaam771 replied:
<img width="1055" alt="image" src="https://user-images.githubusercontent.com/60038592/168225693-5f67b5a0-f01e-45d4-881f-757f63c583de.png">

euny0ung committed:
style: Update Professor

불필요한 코드를 삭제하고 사진 크기를 조정하였습니다.
-Route Article 삭제
-경로 이름을 Prof -> Professor로 수정
-사진 크기를 20% -> 25vh로 수정
#37

euny0ung committed:
refactor: Update Professor
-Box 삭제
-사진 크기 고정
#37

euny0ung committed:
refactor: Update Professor
ProfPrint 함수를 추가하여Typography 태그의 사용을 줄였습니다.

#37

euny0ung committed:
Merge branch 'develop' into feature/professor_page

euny0ung commented:
말해주신 부분들 다 수정하였습니다! 확인 부탁드려요

LimEunSang reviewed (approved):
LGTM!
ProfPrint 함수 정의해서 매개변수 받아 사용하는 것까지 깔끔하게 잘 사용하셨네요! 사진도 고정 크기로 하니까 훨씬 나아보여요. 고생하셨어요!

suinj8 commented on src/App.js:
```diff
@@ -24,13 +25,13 @@ const App = () => {
           <Route path="/location" element={<Location />} />
           <Route path="/history" element={<History />} />
           <Route path="/introduction" element={<Introduction />} />
-          <Route path="/article" element={<Article />} />
```
Article page는 왜 빼신건가용?

euny0ung replied:
push후 충돌이 발생해서 develop이랑 병합하였는데 이 과정에서 빠진것 같아요..! 저번에도 이런 일이 발생해서 직접 수정했는데 혹시 이유를 아시나요??

suinj8 replied:
저도 잘 모르겠네요,,

euny0ung committed:
Update App.js

suinj8 reviewed (approved):
수정사항 확인했습니다.
LGTM

baaaam771 commented on src/pages/member/Professor.js:
```diff
@@ -94,70 +87,63 @@ export default function Professor() {
                 <CardMedia
                   component="img"
                   sx={{
-                    width: "25vh",
```
브라우저 크기에 반응하는 vh, vw 단위가 있었네요 덕분에 알게 됩니다

baaaam771 reviewed (commented):
카드 사이의 간격이 브라우저의 크기에 따라 차이가 많이 생기는 것 같아요

baaaam771 commented:
카드 사이의 간격이 브라우저의 크기에 따라 차이가 많이 생기는 것 같아요
<전체화면>
<img width="1426" alt="image" src="https://user-images.githubusercontent.com/60038592/168739283-7a9b6f35-62ef-42b6-8066-c0ffda4de372.png">
<축소화면>
<img width="485" alt="image" src="https://user-images.githubusercontent.com/60038592/168739422-554e63b1-cae2-4871-a35f-68e206a843c1.png">

euny0ung committed:
refactor: Update Professor

창을 줄였을 때 카드 사이의 간격 수정

euny0ung committed:
refactor: Update Professor

창 줄였을 때 카드 사이의 간격 수정

euny0ung commented:
> 카드 사이의 간격이 브라우저의 크기에 따라 차이가 많이 생기는 것 같아요

my를 md: "2%", xs: "5%" 로 변경하였습니다! 첫번째 사진은 화면 크기가 딱 xs만큼 됐을때의 간격이고 두번째 사진은 최대로 줄였을 때 간격인데 괜찮나요?? 

![중간줄임](https://user-images.githubusercontent.com/56199780/168979260-7828d050-ecb8-4e56-aeac-e998e086f0d0.PNG)
![최대줄임](https://user-images.githubusercontent.com/56199780/168979279-b19ada5a-01a5-4657-9294-b35d4ee89e25.PNG)

baaaam771 reviewed (approved):
넵 좋습니다 수정하느라 고생하셨습니다!

euny0ung committed:
refactor: Update Professor

card 사이의 my를 퍼센트에서 고정값으로 변경

LimEunSang committed:
Merge branch 'develop' into feature/professor_page
